fhir/src/main/java/gov/cms/ab2d/fhir/IdentifierUtils.java

```diff
@@ -100,32 +161,113 @@ private static String getValue(ICompositeType identifier) {
         return (String) Versions.invokeGetMethod(identifier, "getValue");
     }
 
-    private static boolean isBeneficiaryId(ICompositeType identifier) {
-        String system = getSystem(identifier);
-        String value = getValue(identifier);
-        if (StringUtils.isAnyBlank(system, value)) {
-            return false;
+    /**
+     * From a FHIR identifier include MBI information, get the currency information
+     *
+     * @param identifier - the identifier
+     * @return - the currency information
+     */
+    private static PatientIdentifier.Currency getCurrencyMbi(ICompositeType identifier) {
+        Optional<IBaseExtension> currencyExtension = getCurrencyExtension(identifier);
+        if (currencyExtension.isEmpty()) {
+            return PatientIdentifier.Currency.UNKNOWN;
+        }
+        Object code = Versions.invokeGetMethod(currencyExtension.get(), "getValue");
+        String codeValue = (String) Versions.invokeGetMethod(code, "getCode");
+        switch (codeValue) {
+            case HISTORIC_MBI:
+                return PatientIdentifier.Currency.HISTORIC;
+            case CURRENT_MBI:
+                return PatientIdentifier.Currency.CURRENT;
+            default:
+                return PatientIdentifier.Currency.UNKNOWN;
         }
-        return system.equalsIgnoreCase(BENEFICIARY_ID);
     }
 
-    private static boolean isMatchingMbi(ICompositeType identifier, String historic) {
+    /**
+     * Get the currency information for CARIN standard MBI identifier
+     *
+     * @param identifier - the FHIR identifier
+     * @return the currency information
+     */
+    private static PatientIdentifier.Currency getCurrencyMbiStandard(ICompositeType identifier) {
+        Object period = Versions.invokeGetMethod(identifier, "getPeriod");
+        Date start = null;
+        Date end = null;
+        if (period != null) {
+            start = (Date) Versions.invokeGetMethod(period, "getStart");
+            end = (Date) Versions.invokeGetMethod(period, "getEnd");
+        }
+        if (period != null && (start != null || end != null)) {
+            Date now = new Date();
+            if ((start == null || now.getTime() > start.getTime()) && (end == null || end.getTime() > now.getTime())) {
+                return PatientIdentifier.Currency.CURRENT;
+            } else {
+                return PatientIdentifier.Currency.HISTORIC;
+            }
+        } else {
```

Comment:
This else impairs readability since the previous if always results in a return.

Perhaps refactoring the contents of the else to another method would help in readability and do more to preserve a level of abstraction.